pkg/lightning/restore/restore.go

```diff
@@ -1798,26 +1760,58 @@ func (rc *Controller) isLocalBackend() bool {
 // 3. Lightning configuration
 // before restore tables start.
 func (rc *Controller) preCheckRequirements(ctx context.Context) error {
-	if !rc.cfg.App.CheckRequirements {
-		log.L().Info("skip pre check due to user requirement")
-		return nil
-	}
 	if err := rc.ClusterIsAvailable(ctx); err != nil {
 		return errors.Trace(err)
 	}
 
 	if err := rc.StoragePermission(ctx); err != nil {
 		return errors.Trace(err)
 	}
-
-	if err := rc.ClusterResource(ctx); err != nil {
-		return errors.Trace(err)
+	if err := rc.metaMgrBuilder.Init(ctx); err != nil {
+		return err
 	}
+	taskExist := false
 
 	if rc.isLocalBackend() {
-		if err := rc.LocalResource(ctx); err != nil {
+		source, err := rc.EstimateSourceData(ctx)
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		// disable some pd schedulers
+		pdController, err := pdutil.NewPdController(ctx, rc.cfg.TiDB.PdAddr,
+			rc.tls.TLSConfig(), rc.tls.ToPDSecurityOption())
```

Comment:
wrong comment? 😅 no schedulers are disabled here, the comment should be kept after line 1206.

Reply:
Wrong comment, I will remove it.